### PR TITLE
Support printing sensors using `ign model`

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🎉 New feature

## Summary

Print some sensor information using `ign model`. Output is currently limited to altimeter, imu, camera, and rgbd cameras.

## Test it

1. `ign gazebo -v4 sensors.sdf`
2. `ign model -m sensors_box`

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: src/cmd/ModelCommandAPI.cc [hunk 1]
```diff
@@ -117,19 +130,328 @@ std::string entityInfo(const std::string &_name,
...
+  std::cout << std::string(_spaces, ' ') << "- Vertical position noise:\n";
+  printNoise(altimeter->VerticalPositionNoise(), _spaces + 2);
+
+  std::cout << std::string(_spaces, ' ') << "- Vertical velocity noise:\n";
+  printNoise(altimeter->VerticalVelocityNoise(), _spaces + 2);
+}
+
+//////////////////////////////////////////////////
+/// \brief Print info about an SDF camera.
+/// \param[in] _camera The camera to output
```
Comment (ahcorde) [inline src/cmd/ModelCommandAPI.cc:212]:
```suggestion
/// \param[in] _camera The camera to output
/// \param[in] _spaces Number of spaces to indent for every line
```

File: src/cmd/ModelCommandAPI.cc [hunk 2]
```diff
@@ -117,19 +130,328 @@ std::string entityInfo(const std::string &_name,
...
+  std::cout << std::string(_spaces, ' ')
+    << "- Lens intrinsics skew: " << _camera->LensIntrinsicsSkew()
+    << std::endl;
+  std::cout << std::string(_spaces, ' ')
+    << "- Visibility mask: " << _camera->VisibilityMask()
+    << std::endl;
+}
+
+//////////////////////////////////////////////////
+/// \brief Print info about a camera sensor.
```
Comment (ahcorde) [inline src/cmd/ModelCommandAPI.cc:336]:
```suggestion
/// \brief Print info about a RGBD camera sensor.
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/cmd/ModelCommandAPI.cc (hunk 2 above)
Comment (nkoenig) [reply src/cmd/ModelCommandAPI.cc]: 07abdbf

File: src/cmd/ModelCommandAPI.cc (hunk 1 above)
Comment (nkoenig) [reply src/cmd/ModelCommandAPI.cc]: 07abdbf

File: src/ModelCommandAPI_TEST.cc [hunk 3]
```diff
@@ -369,19 +369,171 @@ TEST(ModelCommandAPI, Commands)
...
   }
 }
 
+/////////////////////////////////////////////////
+// Tests `ign model -s` command with an altimeter.
+TEST(ModelCommandAPI, AltimeterSensor)
+{
+  ignition::gazebo::ServerConfig serverConfig;
+  // Using an static model to avoid any movements in the simulation.
+  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
```
Comment (ahcorde) [inline src/ModelCommandAPI_TEST.cc:391]: ignition::common::JointPaths

File: src/ModelCommandAPI_TEST.cc [hunk 4]
```diff
@@ -369,19 +369,171 @@ TEST(ModelCommandAPI, Commands)
...
+  }
+}
+
+/////////////////////////////////////////////////
+// Tests `ign model -s` command with a magnetometer.
+TEST(ModelCommandAPI, MagnetometerSensor)
+{
+  ignition::gazebo::ServerConfig serverConfig;
+  // Using an static model to avoid any movements in the simulation.
+  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
```
Comment (ahcorde) [inline src/ModelCommandAPI_TEST.cc:440]: ignition::common::JointPaths

File: src/ModelCommandAPI_TEST.cc [hunk 5]
```diff
@@ -369,19 +369,171 @@ TEST(ModelCommandAPI, Commands)
...
+  }
+}
+
+/////////////////////////////////////////////////
+// Tests `ign model -s` command with an rgbd camera.
+TEST(ModelCommandAPI, RgbdCameraSensor)
+{
+  ignition::gazebo::ServerConfig serverConfig;
+  // Using an static model to avoid any movements in the simulation.
+  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
```
Comment (ahcorde) [inline src/ModelCommandAPI_TEST.cc:473]: ignition::common::JointPaths

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/ModelCommandAPI_TEST.cc (hunk 4 above)
Comment (nkoenig) [reply src/ModelCommandAPI_TEST.cc]: eae7bcd

File: src/ModelCommandAPI_TEST.cc (hunk 5 above)
Comment (nkoenig) [reply src/ModelCommandAPI_TEST.cc]: eae7bcd

File: src/ModelCommandAPI_TEST.cc (hunk 3 above)
Comment (nkoenig) [reply src/ModelCommandAPI_TEST.cc]: eae7bcd

File: src/ModelCommandAPI_TEST.cc [hunk 6]
```diff
@@ -89,8 +89,9 @@ TEST(ModelCommandAPI, Commands)
 {
   ignition::gazebo::ServerConfig serverConfig;
   // Using an static model to avoid any movements in the simulation.
-  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
-                          "/test/worlds/static_diff_drive_vehicle.sdf");
+  serverConfig.SetSdfFile(
+      ignition::common::joinPaths(std::string(PROJECT_SOURCE_PATH) +
+                          "/test/worlds/static_diff_drive_vehicle.sdf"));
```
Comment (ahcorde) [inline src/ModelCommandAPI_TEST.cc:94]:
```suggestion
      ignition::common::joinPaths(std::string(PROJECT_SOURCE_PATH), "test", "worlds", "static_diff_drive_vehicle.sdf"));
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
Same in the other JoinPaths

File: src/ModelCommandAPI_TEST.cc (hunk 6 above)
Comment (nkoenig) [reply src/ModelCommandAPI_TEST.cc]: Right, thanks. Done.

File: src/ModelCommandAPI_TEST.cc [hunk 7]
```diff
@@ -90,8 +90,8 @@ TEST(ModelCommandAPI, Commands)
   ignition::gazebo::ServerConfig serverConfig;
   // Using an static model to avoid any movements in the simulation.
   serverConfig.SetSdfFile(
-      ignition::common::joinPaths(std::string(PROJECT_SOURCE_PATH) +
-                          "/test/worlds/static_diff_drive_vehicle.sdf"));
+      ignition::common::joinPaths(std::string(PROJECT_SOURCE_PATH),
+        "/test/worlds/static_diff_drive_vehicle.sdf"));
```
Comment (ahcorde) [inline src/ModelCommandAPI_TEST.cc]: still need to split this with comas:
```
"test", "worlds", "static_diff_drive_vehicle.sdf"));
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/ModelCommandAPI_TEST.cc (hunk 7 above)
Comment (nkoenig) [reply src/ModelCommandAPI_TEST.cc]: Yup, thanks. Done.

**Review by ahcorde — APPROVED** [review]

**Pull request merged**